src/main/java/org/elasticsearch/action/admin/cluster/snapshots/create/TransportCreateSnapshotAction.java

```diff
@@ -65,7 +65,7 @@ protected CreateSnapshotResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(CreateSnapshotRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
I see why this is is confusing. But if we move this to METADATA_READ, will we be able to make a snapshot when the cluster is in read_only mode? If so, I would consider changing to METADATA_READ for now, which fixes #8102 too , add a comment that explains why (pretty much Igor's comment above) and next step could be to move snapshot out of metadata even?